internal/service/encryptionatrest/model_encryption_at_rest.go

```diff
@@ -0,0 +1,120 @@
+package encryptionatrest
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"go.mongodb.org/atlas-sdk/v20231115002/admin"
+)
+
+func NewTfEncryptionAtRestRSModel(ctx context.Context, projectID string, encryptionResp *admin.EncryptionAtRest) *TfEncryptionAtRestRSModel {
+	return &TfEncryptionAtRestRSModel{
+		ID:                   types.StringValue(projectID),
+		ProjectID:            types.StringValue(projectID),
+		AwsKmsConfig:         NewTFAwsKmsConfig(ctx, encryptionResp.AwsKms),
+		AzureKeyVaultConfig:  NewTFAzureKeyVaultConfig(ctx, encryptionResp.AzureKeyVault),
+		GoogleCloudKmsConfig: NewTFGcpKmsConfig(ctx, encryptionResp.GoogleCloudKms),
+	}
+}
+
+func NewTFAwsKmsConfig(ctx context.Context, awsKms *admin.AWSKMSConfiguration) []TfAwsKmsConfigModel {
```

Comment:
some of this methods had extra parameters which were not used. I removed them to clean up the code.